src/js/controller/instream-adapter.js

```diff
@@ -276,6 +276,14 @@ var InstreamAdapter = function(_controller, _model, _view) {
         }
     };
 
+    this.replacePlaylistItem = function(url) {
+        const item = _model.get('playlistItem');
+        item.sources[0].file = url;
```

Comment:
What do you think about replacing `sources` with a single source?

`item.sources = [Object.assign(item.sources[0], {file: url})]`

Reply:
If we want to leave the playlist unmodified, then we need to set 'playlistItem'. That would ensure that if we left this playlist item, and came back to it, that the same replacement would happen - otherwise we would feed the modified url back to the SDK.

Reply:
Thank you for looking into this! I wanted to run by you on this actually. If we request stream to the SDK on every playlistItem, it would re-request stream when we do set 'playlistItem'. Do you have any thoughts on how we could prevent that?

Reply:
If you don't modify the original object using my example above, then you can use `allSources`. We should be using `allSources` to find the correct media type to pass to the SDK anyway.